src/proxy/ILock.ts

```diff
@@ -0,0 +1,73 @@
+export interface Lock {
+
+    /**
+     * Acquires the lock, waiting indefinitely if necessary.
+     * The returned promise will be resolved as soon as the lock becomes available.
+     * After this method is called the user must call `unlock` to release the lock.
+     */
+    lock(): Promise<void>;
+
+    /**
+     * Leases the lock for a specified period of time, waiting indefinitely if necessary.
+     * The returned promise will be resolved as soon as the lock becomes available.
+     * The lock will be held for the specified amount of time and then released automatically.
+     *
+     * @param leaseMillis period of time in milliseconds for which the lock should be held.
+     */
+    lease(leaseMillis: number): Promise<void>;
```

Comment:
Why do we have `lease` as a seperate method? We can put `leaseMillis` into `lock()` method as an optional argument. This is also how other clients do it.